core/metaobjectrepository.cpp

```diff
@@ -183,6 +185,39 @@ void MetaObjectRepository::initIOTypes()
     MO_ADD_METAOBJECT1(QSaveFile, QFileDevice);
 }
 
+Q_DECLARE_METATYPE(const QObject*)
+Q_DECLARE_METATYPE(void**)
+
+void MetaObjectRepository::initQEventTypes()
+{
+    MetaObject *mo = nullptr;
+    MO_ADD_METAOBJECT0(QEvent);
+    MO_ADD_PROPERTY_RO(QEvent, isAccepted);
+    MO_ADD_PROPERTY_RO(QEvent, spontaneous);
+    MO_ADD_PROPERTY_RO(QEvent, type);
+
+    MO_ADD_METAOBJECT1(QTimerEvent, QEvent);
+    MO_ADD_PROPERTY_RO(QTimerEvent, timerId);
+
+    MO_ADD_METAOBJECT1(QChildEvent, QEvent);
+    MO_ADD_PROPERTY_RO(QChildEvent, child);
+    MO_ADD_PROPERTY_RO(QChildEvent, added);
+    MO_ADD_PROPERTY_RO(QChildEvent, polished);
+    MO_ADD_PROPERTY_RO(QChildEvent, removed);
+
+    MO_ADD_METAOBJECT1(QDynamicPropertyChangeEvent, QEvent);
+    MO_ADD_PROPERTY_RO(QDynamicPropertyChangeEvent, propertyName);
+
+    MO_ADD_METAOBJECT1(QDeferredDeleteEvent, QEvent);
+    MO_ADD_PROPERTY_RO(QDeferredDeleteEvent, loopLevel);
+
+    MO_ADD_METAOBJECT1(QMetaCallEvent, QEvent);
+    MO_ADD_PROPERTY_RO(QMetaCallEvent, id);
+//    MO_ADD_PROPERTY_RO(QMetaCallEvent, sender);  // problematic because type is const QObject*
+    MO_ADD_PROPERTY_RO(QMetaCallEvent, signalId);
+//    MO_ADD_PROPERTY_RO(QMetaCallEvent, args);  // unusable atm because type is void**
```

Comment:
Right, and this can never work as-is, this will need custom decoding code for args. So we can remove that here, as well as the metatype declaration for void**.